### ISSUE TITLE
Porting twisted.application.app to python3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6914]
|<img alt="realcr's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| realcr reported|
|-|-|
|Trac ID|trac#6914|
|Type|enhancement|
|Created|2014-01-18 08:54:33Z|
|Branch|https://github.com/twisted/twisted/tree/twapp-py3-6914-2|

app.py will support both python2.7 and python3.3 syntax.

Sent patch contains the following:
- Fixing except statements.
- Adding braces to the print function.
- Relevant from __future__ import ... statement.

Attachments:

* [application_app_py3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/27e/27eedc1e245d7253a951f5960a239661ef2556fa/e654246ec2a728e6c88a0c8a6d858c0a42951142.patch) (3293 bytes) - added by realcr on 2014-01-18 08:54:56Z - Syntax changes to work on python3.3

<details><summary>Searchable metadata</summary>

```
trac-id__6914 6914
type__enhancement enhancement
reporter__realcr realcr
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_twapp_py3_6914_2 branches/twapp-py3-6914-2
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1390035273000000 1390035273000000
changetime__1435847550480221 1435847550480221
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to realcr|
|-|-|

Comments for this are basically the same as for [#6910](https://github.com/twisted/twisted/issues/6910).

Since `print` usage is actually sort of difficult to test, I suspect that some of the changes here are to untested code.  It may be necessary to add some new tests for this code.

And since `print` usage is difficult to test, you may want to re-arrange things a bit - for example, you could have the APIs accept a new, optional `output` argument that defaults to `sys.stdout` and then uses `output.write` instead of `print`.  Then the unit tests can pass a `StringIO` and easily make assertions about what was written.

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [45157]) Branching to twapp-py3-6914.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [45164]) Branching to twapp-py3-6914-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl removed owner|
|-|-|

I'm pretty happy with this now. The coverage is a tad spotty, but they're used by as of yet unported bits of Twisted, as well as twistd itself.

Builders looking green, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

For twisted/test/test_twistd.py

```
1014+        data = open(config["profile"]).read()
```

maybe you can update it to use the context manager.

------

Why is twisted.scripts.twistd in almostModules when its test is in testModules ?

Please check my comments and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

Hi Adi,

Thanks for the catch, fixed.

As for twisted.scripts.twistd, it's not much use without the executable in the setup, which will be a different ticket.

Will rerun builders to make sure after that change, and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45180]) Merge twapp-py3-6914-2: Port twisted.application.app to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#6914](https://github.com/twisted/twisted/issues/6914)
